epidose/device/wps_scanner_d.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+#
+# Check for available networks and connect to the closest one using the WPS PBS
+#
+# Copyright 2020 Diomidis Spinellis
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+export APP_NAME=wps_scanner_d
+
+# Pick up utility functions relative to the script's source code
+UTIL="$(dirname "$0")/util.sh"
+
+# Source common functionality (logging, WiFi)
+# shellcheck source=epidose/device/util.sh
+SERVER_URL=none . "$UTIL"
+
+wps_connect()
+{
+  # Turn on the orange blinking of the WPS LED
+  #run_python device_io --orange-on
+
+  network_status=$(ifconfig wlan0 | grep "inet")
+  
+  if [ -z "$network_status" ] ; then
+    log "Iniating WPS connection"
+    
+    # Connected to the network that supports WPS and has the lowest signal level
+    wpa_cli scan_results | grep WPS | sort -r -k3 | tail -1 | awk '{print $1}' | xargs wpa_cli wps_pbc
+    exit_code=$?
+    
+    # Obtain dynamic IP address
+    dhclient wlan0
+  fi
+
+  # Turn off the orange blinking of the WPS LED
+  #run_python device_io --orange-off
+  
+  return $exit_code
+}
+
+# Wait for WPS button to be pressed
+while : ; do
+  while ! wps_connect; do
```

Comment:
I don't think you need a second loop.  Use `if`?

Reply:
True. Replaced.